src/DiscoveryAnnouncer.coffee

```diff
@@ -1,83 +1,117 @@
-Promise        = require "bluebird"
-Errors         = require "./Errors"
+Promise = require "bluebird"
+Errors = require "./Errors"
 RequestPromise = require "./RequestPromise"
-Utils          = require "./Utils"
-_              = require "lodash"
+Utils = require "./Utils"
+_ = require "lodash"
+
+
+# this is now static because we have multiple discover annoucers per annouce region!
```

Comment:
I imagine it could be cleaner to have an `DiscoveryAnnouncer` per region, that is, have multiple objects held in an list ... you would not have to change the code here as much and it would be easier to reason about and test.